### PR TITLE
Retrieve vault token via IAM role access in AWS CPs

### PR DESCRIPTION
**Pull request by corest**
Towards https://github.com/giantswarm/giantswarm/issues/12862

### CODE REVIEW

File: templates/master.yaml.tmpl [hunk 1]
```diff
@@ -805,7 +805,7 @@ systemd:
           -e VAULT_ADDR\
           -e CLUSTER_NAME\
           {{.DockerRegistry}}/giantswarm/vault:1.6.1\
-          login -method=aws role=$${CLUSTER_NAME}-master -format=json | jq -r .auth.client_token);\
+          login -method=aws role=$${CLUSTER_NAME}-master region={{ .AWSRegion }} -format=json | jq -r .auth.client_token);\
```
Comment (kopiczko) [inline templates/master.yaml.tmpl:808]: Looks ok but how was it working so far? Also may be a step backwards when it comes to multi-region support if we are planning to ever do that.

**Review by kopiczko — APPROVED** [review]

File: templates/master.yaml.tmpl [hunk 2]
```diff
@@ -774,13 +776,49 @@ systemd:
...
+          export vault_token=$(\
+          docker run --rm -i\
+          -v /etc/ssl/certs/ca-certificates.crt:/etc/ssl/certs/ca-certificates.crt\
+          --net host\
+          --privileged=true\
+          -e VAULT_ADDR\
+          -e CLUSTER_NAME\
+          {{.DockerRegistry}}/giantswarm/vault:1.6.1\
+          login -method=aws role=$${CLUSTER_NAME}-master region={{ .AWSRegion }} -format=json | jq -r .auth.client_token);\
+          echo "VAULT_TOKEN=$${vault_token}" > /etc/tokens/node;\
```
Comment (calvix) [inline templates/master.yaml.tmpl:809]: when the aws command fails the   main  `ExecStart` command exits with 0 (due to the last command  doing just echo) thus silently falining

Comment (kopiczko) [reply templates/master.yaml.tmpl:809]: We can replace with `/bin/bash -Oe -c '\`

File: templates/master.yaml.tmpl (hunk 1 above)
Comment (corest) [reply templates/master.yaml.tmpl:808]: It is a new service. We didn't have it before. Also multi-region support has nothing to do with this service.

Comment (kopiczko) [reply templates/master.yaml.tmpl:808]: Do you know what is this region used for? I quickly checked https://www.vaultproject.io/docs/auth/aws but it didn't make much sense to me. I thought IAM is global service.

Comment (corest) [reply templates/master.yaml.tmpl:808]: China uses a different STS endpoint, therefore, requires setting a corresponding region explicitly

Comment (corest) [reply templates/master.yaml.tmpl:808]: And there is example on the page you've referenced

```
The region used defaults to us-east-1, but you can specify a custom region like so:

$ vault login -method=aws region=us-west-2 role=dev-role-iam

When using a custom region, be sure the designated region corresponds to that of the STS endpoint you're using.
```

**Pull request merged**